File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_cpp/MovieTicketDB.cpp
#include <sqlite3.h>
#include <iostream>
#include <vector>
#include <string>

class MovieTicketDB {
public:
    struct Ticket {
        int id;
        std::string movieName;
        std::string theaterName;
        std::string seatNumber;
        std::string customerName;

        Ticket(int id, const std::string& movieName, const std::string& theaterName,
               const std::string& seatNumber, const std::string& customerName)
            : id(id), movieName(movieName), theaterName(theaterName),
              seatNumber(seatNumber), customerName(customerName) {}

        int getId() const { return id; }
        std::string getMovieName() const { return movieName; }
        std::string getTheaterName() const { return theaterName; }
        std::string getSeatNumber() const { return seatNumber; }
        std::string getCustomerName() const { return customerName; }
    };

    MovieTicketDB(const std::string& dbName) : db(nullptr) {
        if (sqlite3_open(dbName.c_str(), &db) != SQLITE_OK) {
            std::cerr << "Cannot open database: " << sqlite3_errmsg(db) << std::endl;
            sqlite3_close(db);
            db = nullptr;
            return;
        }
        create_table();
    }

    ~MovieTicketDB() {
        close_connection();
    }

    void close_connection() {
        if (db) {
            if (sqlite3_close(db) != SQLITE_OK) {
                std::cerr << "Failed to close database: " << sqlite3_errmsg(db) << std::endl;
            }
            db = nullptr;
        }
    }

    void insert_ticket(const std::string& movieName, const std::string& theaterName,
                      const std::string& seatNumber, const std::string& customerName) {
        if (!db) return;

        const char* sql = "INSERT INTO tickets (movie_name, theater_name, seat_number, customer_name) VALUES (?, ?, ?, ?)";
        sqlite3_stmt* stmt = nullptr;

        if (sqlite3_prepare_v2(db, sql, -1, &stmt, nullptr) != SQLITE_OK) {
            std::cerr << "Failed to prepare statement: " << sqlite3_errmsg(db) << std::endl;
            return;
        }

        sqlite3_bind_text(stmt, 1, movieName.c_str(), -1, SQLITE_STATIC);
        sqlite3_bind_text(stmt, 2, theaterName.c_str(), -1, SQLITE_STATIC);
        sqlite3_bind_text(stmt, 3, seatNumber.c_str(), -1, SQLITE_STATIC);
        sqlite3_bind_text(stmt, 4, customerName.c_str(), -1, SQLITE_STATIC);

        if (sqlite3_step(stmt) != SQLITE_DONE) {
            std::cerr << "Failed to execute statement: " << sqlite3_errmsg(db) << std::endl;
        }

        sqlite3_finalize(stmt);
    }

    std::vector<Ticket> searchTicketsByCustomer(const std::string& customerName) {
        std::vector<Ticket> tickets;
        if (!db) return tickets;

        const char* sql = "SELECT * FROM tickets WHERE customer_name = ?";
        sqlite3_stmt* stmt = nullptr;

        if (sqlite3_prepare_v2(db, sql, -1, &stmt, nullptr) != SQLITE_OK) {
            std::cerr << "Failed to prepare statement: " << sqlite3_errmsg(db) << std::endl;
            return tickets;
        }

        sqlite3_bind_text(stmt, 1, customerName.c_str(), -1, SQLITE_STATIC);

        int rc;
        while ((rc = sqlite3_step(stmt)) == SQLITE_ROW) {
            int id = sqlite3_column_int(stmt, 0);
            const unsigned char* movie = sqlite3_column_text(stmt, 1);
            const unsigned char* theater = sqlite3_column_text(stmt, 2);
            const unsigned char* seat = sqlite3_column_text(stmt, 3);
            const unsigned char* customer = sqlite3_column_text(stmt, 4);

            tickets.emplace_back(id,
                std::string(reinterpret_cast<const char*>(movie)),
                std::string(reinterpret_cast<const char*>(theater)),
                std::string(reinterpret_cast<const char*>(seat)),
                std::string(reinterpret_cast<const char*>(customer)));
        }

        if (rc != SQLITE_DONE) {
            std::cerr << "Error during query: " << sqlite3_errmsg(db) << std::endl;
        }

        sqlite3_finalize(stmt);
        return tickets;
    }

    void delete_ticket(int ticketId) {
        if (!db) return;

        const char* sql = "DELETE FROM tickets WHERE id = ?";
        sqlite3_stmt* stmt = nullptr;

        if (sqlite3_prepare_v2(db, sql, -1, &stmt, nullptr) != SQLITE_OK) {
            std::cerr << "Failed to prepare statement: " << sqlite3_errmsg(db) << std::endl;
            return;
        }

        sqlite3_bind_int(stmt, 1, ticketId);

        if (sqlite3_step(stmt) != SQLITE_DONE) {
            std::cerr << "Failed to execute statement: " << sqlite3_errmsg(db) << std::endl;
        }

        sqlite3_finalize(stmt);
    }

private:
    sqlite3* db;

    void create_table() {
        if (!db) return;

        const char* sql = "CREATE TABLE IF NOT EXISTS tickets ("
            "id INTEGER PRIMARY KEY AUTOINCREMENT, "
            "movie_name TEXT, "
            "theater_name TEXT, "
            "seat_number TEXT, "
            "customer_name TEXT)";
        char* errMsg = nullptr;

        if (sqlite3_exec(db, sql, nullptr, nullptr, &errMsg) != SQLITE_OK) {
            std::cerr << "SQL error: " << errMsg << std::endl;
            sqlite3_free(errMsg);
        }
    }
};

File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/MovieTicketDB.py
import sqlite3
import traceback


class MovieTicketDB:
    class Ticket:
        def __init__(self, id_: int, movie_name: str, theater_name: str, seat_number: str, customer_name: str):
            self.id = id_
            self.movie_name = movie_name
            self.theater_name = theater_name
            self.seat_number = seat_number
            self.customer_name = customer_name

        def get_id(self) -> int:
            return self.id

        def get_movie_name(self) -> str:
            return self.movie_name

        def get_theater_name(self) -> str:
            return self.theater_name

        def get_seat_number(self) -> str:
            return self.seat_number

        def get_customer_name(self) -> str:
            return self.customer_name

    def __init__(self, db_name: str):
        try:
            self.connection = sqlite3.connect(db_name)
            self._create_table()
        except sqlite3.Error:
            traceback.print_exc()
            self.connection = None

    def _create_table(self):
        sql = """CREATE TABLE IF NOT EXISTS tickets (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            movie_name TEXT,
            theater_name TEXT,
            seat_number TEXT,
            customer_name TEXT
        )"""
        try:
            with self.connection:
                self.connection.execute(sql)
        except sqlite3.Error:
            traceback.print_exc()

    def insert_ticket(self, movie_name: str, theater_name: str, seat_number: str, customer_name: str):
        sql = "INSERT INTO tickets (movie_name, theater_name, seat_number, customer_name) VALUES (?, ?, ?, ?)"
        try:
            with self.connection:
                self.connection.execute(sql, (movie_name, theater_name, seat_number, customer_name))
        except sqlite3.Error:
            traceback.print_exc()

    def search_tickets_by_customer(self, customer_name: str):
        sql = "SELECT * FROM tickets WHERE customer_name = ?"
        tickets = []
        try:
            with self.connection:
                cursor = self.connection.execute(sql, (customer_name,))
                for row in cursor:
                    ticket = self.Ticket(
                        id_=row[0],
                        movie_name=row[1],
                        theater_name=row[2],
                        seat_number=row[3],
                        customer_name=row[4]
                    )
                    tickets.append(ticket)
        except sqlite3.Error:
            traceback.print_exc()
        return tickets

    def delete_ticket(self, ticket_id: int):
        sql = "DELETE FROM tickets WHERE id = ?"
        try:
            with self.connection:
                self.connection.execute(sql, (ticket_id,))
        except sqlite3.Error:
            traceback.print_exc()

    def close(self):
        try:
            if self.connection is not None:
                self.connection.close()
        except sqlite3.Error:
            traceback.print_exc()